### PR TITLE
Normative: Add Nullish Coalescing

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
This change adds the relevant updates for the stage 3 [Nullish Coalescing](https://github.com/tc39/proposal-nullish-coalescing) proposal. The intended changes can be viewed at https://tc39.es/proposal-nullish-coalescing/.

The Test262 PR is available at https://github.com/tc39/test262/pull/2402

### CODE REVIEW

**Comment by anba** [comment]
The `?In` -> `+In` change for ConditionalExpression is incorrect.

File: spec.html [hunk 1]
```diff
@@ -14084,8 +14108,8 @@ <h1>Conditional Operator ( `? :` )</h1>
     <h2>Syntax</h2>
     <emu-grammar type="definition">
       ConditionalExpression[In, Yield, Await] :
-        LogicalORExpression[?In, ?Yield, ?Await]
-        LogicalORExpression[?In, ?Yield, ?Await] `?` AssignmentExpression[+In, ?Yield, ?Await] `:` AssignmentExpression[?In, ?Yield, ?Await]
+        LogicalExpression[?In, ?Yield, ?Await]
+        LogicalExpression[?In, ?Yield, ?Await] `?` AssignmentExpression[+In, ?Yield, ?Await] `:` AssignmentExpression[+In, ?Yield, ?Await]
```
Comment (jridgewell) [inline spec.html]: @anba means this one.

Comment (DanielRosenwasser) [reply spec.html]: Oof, thank you both.

**Comment by ljharb** [comment]
@DanielRosenwasser please rebase this PR, and either link to the test262 tests PR, or put one up, prior to the December meeting. Thanks!

**Comment by bakkot** [comment]
@DanielRosenwasser I think you messed up the rebase?

**Comment by DanielRosenwasser** [comment]
I did! But I unbroke it and after a little bit of fiddling and waiting, GitHub is less angry with me and let me reopen the PR. 🙂

**Comment by DanielRosenwasser** [comment]
Also, the PR now links to https://github.com/tc39/test262/pull/2402, the test262 additions PR.

**Comment by claudepache** [comment]
As noted in https://github.com/tc39/proposal-nullish-coalescing/issues/59, the token `??` must be added to the [Punctuator production](https://tc39.es/ecma262/#prod-Punctuator) of the Lexical grammar.

**Comment by ljharb** [comment]
@claudepache done.

**Pull request merged**